### PR TITLE
Added a test to validate json

### PR DESCRIPTION
**Pull request by DevOrc**
Fixes part one of #75.

 Question: Should this test be placed in a different module? For example have a source_tests module and generic_tests module

### CODE REVIEW

**Comment by steveklabnik** [comment]
Thank you! This looks great.

CI failed due to rustfmt issues.

> Question: Should this test be placed in a different module? For example have a source_tests module and generic_tests module

Yeah, let's put this in a `tests/validation.rs`.

File: tests/source.rs [hunk 1]
```diff
@@ -314,4 +316,42 @@ mod tests {
...
+
+        let result = build(&config, &["json"]);
+
+        if let Err(ref e) = result {
+            panic!("Error: {}", e);
+        }
+
+        let path = Path::new("example/target/doc/data.json");
+
+        let mut file = match File::open(&path) {
```
Comment (steveklabnik) [inline tests/source.rs]: this match could be `unwrap_or_else` like above

File: tests/source.rs [hunk 2]
```diff
@@ -314,4 +316,42 @@ mod tests {
...
+            Err(e) => panic!("couldn't open data.json: {}", e),
+            Ok(file) => file,
+        };
+
+        let mut s = String::new();
+        let result = file.read_to_string(&mut s);
+
+        if let Err(ref e) = result {
```
Comment (steveklabnik) [inline tests/source.rs]: this could also be `unwrap_or_else` on the `read_to_string` call

File: tests/source.rs [hunk 3]
```diff
@@ -314,4 +316,42 @@ mod tests {
...
+            panic!("Error: {}", e);
+        }
+
+        let result = JsonApiDocument::from_str(&s);
+        let doc = match result{
```
Comment (steveklabnik) [inline tests/source.rs]: same here

File: tests/source.rs [hunk 4]
```diff
@@ -314,4 +316,42 @@ mod tests {
...
+            Err(e) => panic!("Error: {}", e),
+            Ok(doc) => doc
+        };
+
+        match doc.validate() {
```
Comment (steveklabnik) [inline tests/source.rs]: and here 😄

Comment (DevOrc) [inline tests/source.rs]: How would unwrap_or_else work here? It should return none if no errors are found.

File: tests/validation.rs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        let mut s = String::new();
+        file.read_to_string(&mut s).unwrap_or_else(|err| {
+            panic!("Error: {}", err);
+        });
+
+        let doc = JsonApiDocument::from_str(&s).unwrap_or_else(|err| {
+            panic!("Error: {}", err);
+        });
+
+        match doc.validate() {
```
Comment (DevOrc) [inline tests/validation.rs]: How would unwrap_or_else work here? It should return none if no errors are found.

Comment (steveklabnik) [inline tests/validation.rs]: oops, you're right!

**Comment by steveklabnik** [comment]
@DevOrc one other thing; it looks like you may not have set up your git properly to link to github 😄

File: tests/validation.rs (hunk 5 above)
Comment (DevOrc) [inline tests/validation.rs]: I thinks its right now!

**Comment by DevOrc** [comment]
I thinks its right now! Do I have to recommit it to update the authorship?

**Comment by mgattozzi** [comment]
This a small nitpick about the title, but using `Add` instead of `Added` makes for a [better commit message](https://chris.beams.io/posts/git-commit/#imperative). Otherwise this is some good code @DevOrc!

**Comment by mgattozzi** [comment]
@steveklabnik not sure if this will conflict with #92 now or not.

**Comment by steveklabnik** [comment]
Thank you!

@mgattozzi we can still use it as a dev-dependency; i'll comment over on that PR

**Pull request merged**